Lib/test/test_largefile.py

```diff
@@ -45,6 +45,13 @@ def tearDownClass(cls):
             raise cls.failureException('File was not truncated by opening '
                                        'with mode "wb"')
 
+    @bigmemtest(size=size, memuse=1, dry_run=False)
```

Comment:
According to tracemalloc, it's more memuse=2. _pyio seems to copy the buffer of 2.5 GB...

So it should be @bigmemtest(size=size, memuse=2, dry_run=False).

Reply:
> _pyio seems to copy the buffer of 2.5 GB...

FYI I looked at _pyio: FileIO.readall() writes the data into a bytearray and then converts the bytearray to a bytes object:

```
        result = bytearray()
        while True:
            ...
                chunk = os.read(self._fd, n)
            ...
            if not chunk:
                break
            result += chunk
        return bytes(result)
```

So yeah, _pyio.FileIO.readall() allocates twice the size of the file content. Sadly, Python doesn't provide a way to convert a bytearray to a bytes object without copying memory. And that's not doable since a bytes object uses a single memory block.